gremlin-dotnet/src/Gremlin.Net/Driver/IResultSet.cs

```diff
@@ -0,0 +1,43 @@
+#region License
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#endregion
+
+using System.Collections.Generic;
+
+namespace Gremlin.Net.Driver
+{
+    /// <summary>
+    /// A IResultSet is returned from the submission of a Gremlin script to the server and represents the results provided by the server
+    /// ResultSet includes enumerable data and status attributes.
+    /// </summary>
+    /// <typeparam name="T">Type of the result elements</typeparam>
+    public interface IResultSet<T> : IReadOnlyCollection<T>
+    {
+        /// <summary>
+        ///  Gets or sets the data from the response
+        /// </summary>
+        IReadOnlyCollection<T> Data {get;}
```

Comment:
I'm very new to this project so excuses if I've missed something, but it seems odd to me that `IResultSet<T>` both is the result (through it's implementation of `IReadOnlyCollection<T>`) and contains it (through the `Data` property).

Reply:
Agree, it looks unnecessary to expose a property.

Reply:
@jorgebay Is there a reason for this being a Collection rather than a List?

Reply:
The reason I implemented IReadOnlyCollection so the existing application do not break. And , we need the Data property to hold the content. I do not want to implement collection to store the the actual data. I can change the member to a List , is that fine ?

A sample example would help if you think I did not get the concerns correctly.

Reply:
@ashwinisingh01 Just make "Data" a private field instead of a property and it's fine.
As far as changing the type to a List, I wouldn't until a core maintainer says it's appropriate, I'm way too new Gremlin/Tinkerpop to say whether that makes sense or not, that's why I directed the question as @jorgebay :)